Code/src/test/java/nl/utwente/group10/haskell/typeparser/TypeBuilderTest.java

```diff
@@ -29,16 +29,22 @@ private void roundtrip(String hs) {
     @Test public void testMaybeMaybe()  { this.roundtrip("Maybe (Maybe a)"); }
     @Test public void testMaybeSink()   { this.roundtrip("Maybe Int -> Maybe [a]"); }
     @Test public void testKitchenSink() { this.roundtrip("[a] -> [b] -> [(a, b)]"); }
+    @Test public void testPrefixUnit()  { this.roundtrip("()"); }
+    @Test public void testPrefixTuple() { this.roundtrip("(,) a"); }
+    @Test public void testPrefixTriple(){ this.roundtrip("(,,) a b"); }
+    @Test public void testPrefixList()  { this.roundtrip("[] a"); }
```

Comment:
Why does roundtrip("[] a") succeed?
I would expect that "[] a" comes out as the special cased haskell type "[a]"